functions/wait/action/kubernetesUpgrade.go

```diff
@@ -5,108 +5,56 @@ import (
 	"k8s.io/apimachinery/pkg/util/wait"
 )
 
-func KubernetesUpgrade(t *testing.T, client *rancher.Client, clusterID string, module string) (done bool, err error) {
-	clusterConfig := new(config.TerratestConfig)
-	framework.LoadConfig("terratest", clusterConfig)
-
-	if module == "aks" || module == "gke" {
-		expectedUpgradedKubernetesVersion := `v` + clusterConfig.UpgradedKubernetesVersion
-		waiting := false
-		waitErr := wait.Poll(100*time.Millisecond, 30*time.Minute, func() (done bool, err error) {
-			cluster, clientErr := client.Management.Cluster.ByID(clusterID)
-			require.NoError(t, err)
-
-			if clientErr != nil {
-				t.Logf("Failed to locate cluster and grab client specs. Error: %v", err)
-				return false, err
-			}
-
-			if cluster.Version.GitVersion == expectedUpgradedKubernetesVersion {
-				t.Logf("Successfully updated kubernetes version to %v", clusterConfig.UpgradedKubernetesVersion)
-				return true, nil
-			}
-
-			if !waiting {
-				t.Logf("Waiting for cluster to upgrade kubernetes version...")
-				waiting = true
-			}
-
-			return false, nil
-		})
-		require.NoError(t, waitErr)
+const (
+	aks        = "aks"
+	eks        = "eks"
+	gke        = "gke"
+	ec2RKE1    = "ec2_rke1"
+	ec2RKE2    = "ec2_rke2"
+	ec2K3s     = "ec2_k3s"
+	linodeRKE1 = "linode_rke1"
+	linodeRKE2 = "linode_rke2"
+	linodeK3s  = "linode_k3s"
+)
 
-		if waitErr != nil {
-			t.Logf("Failed to instantiate kubernetes upgrade wait poll.")
-			return false, waitErr
-		}
+func KubernetesUpgrade(t *testing.T, client *rancher.Client, clusterID string, module string, clusterConfig *config.TerratestConfig) (done bool, err error) {
+	var expectedUpgradedKubernetesVersion string
+
+	switch {
+	case module == aks || module == eks || module == gke:
+		expectedUpgradedKubernetesVersion = `v` + clusterConfig.UpgradedKubernetesVersion
+	case module == ec2RKE1 || module == linodeRKE1:
+		expectedUpgradedKubernetesVersion = clusterConfig.UpgradedKubernetesVersion[:len(clusterConfig.UpgradedKubernetesVersion)-11]
+	case module == ec2RKE2 || module == ec2K3s || module == linodeRKE2 || module == linodeK3s:
+		expectedUpgradedKubernetesVersion = clusterConfig.UpgradedKubernetesVersion
+	default:
+		logrus.Errorf("Unsupported module: %v", module)
 	}
 
-	if module == "ec2_rke1" || module == "linode_rke1" {
-		expectedUpgradedKubernetesVersion := clusterConfig.UpgradedKubernetesVersion[:len(clusterConfig.UpgradedKubernetesVersion)-11]
-		waiting := false
-		waitErr := wait.Poll(100*time.Millisecond, 30*time.Minute, func() (done bool, err error) {
-			cluster, clientErr := client.Management.Cluster.ByID(clusterID)
-			require.NoError(t, err)
-
-			if clientErr != nil {
-				t.Logf("Failed to locate cluster and grab client specs. Error: %v", err)
-				return false, err
-			}
-
-			if cluster.Version.GitVersion == expectedUpgradedKubernetesVersion {
-				return true, nil
-			}
+	waiting := false
+	err = wait.Poll(100*time.Millisecond, 30*time.Minute, func() (done bool, err error) {
```

Comment:
Not sure if we want to do it here but in the tests/v2 we have a defaults.go to store these timeout constants

Reply:
Yeah, I think that would be good as well. There is a side issue where the go versioning needs to be bumped as right now, the go.mod is not getting the latest changes. Once that is resolved, I can address this.